crowbar_framework/app/models/backup.rb

```diff
@@ -164,14 +164,26 @@ def save_archive
       return false
     end
 
-    path.open("wb") do |f|
-      f.write(file.read)
+    meta_file = data.join("meta.yml")
+    unless meta_file.exist?
+      errors.add(:file_content, I18n.t("backups.index.meta_missing"))
+      return false
+    end
+
+    begin
+      meta = YAML.load_file(meta_file)
+    rescue Psych::SyntaxError
+      errors.add(:file_content, I18n.t("backups.index.invalid_file_content"))
+      return false
     end
 
-    meta = YAML.load_file(data.join("meta.yml"))
-    self.version = meta["version"].to_s
+    self.version = meta["version"]
```

Comment:
I think it is expected IIRC because the versions in the model were floats before. but now they are strings. but i will change it back, `to_s` would help when someone (even though unlikely) uploads an old backup which was created with a float version. thanks for catching that